apps/fares/lib/month.ex

```diff
@@ -0,0 +1,119 @@
+defmodule Fares.Month do
+  @moduledoc """
+  Calculates the lowest and highest monthly pass fare for a particular trip.
+  """
+
+  alias Fares.{Fare, Repo}
+  alias Routes.Route
+  alias Schedules.Trip
+  alias Stops.Stop
+
+  @type fare_fn :: (Keyword.t() -> [Fare.t()])
+
+  @spec lowest_pass(
+          Route.t() | Route.id_t(),
+          Trip.t() | Trip.id_t() | nil,
+          Stop.id_t(),
+          Stop.id_t(),
+          fare_fn()
+        ) ::
+          Fare.t() | nil
+  def lowest_pass(route, trip, origin_id, destination_id, fare_fn \\ &Repo.all/1)
```

Comment:
These lowest and highest pass functions (which look good generally speaking) are almost identical save for `Enum.max_by` vs `Enum.min_by`, so I'm wondering if the abstraction is in the wrong place. Maybe we need one method that handles most of the work (`passes_for_route(route, trip, origin_id, destination_id, fare_fn)` maybe, or just expose `get_fares`)

Reply:
Could you please elaborate on this more? I'm not sure I'm following quite yet.

Reply:
Was thinking:

`def get_passes(nil, _, _, _, _), do: nil`
`def get_passes(route_id, trip, origin_id, destination_id, fare_fn) when is_binary(route_id) do`
etc etc etc... 
---eventually it calls `get_fares(route, trip, origin_id, destination_id, fare_fn)` and returns `[Fare.t()] | nil`

then those passes serve as the argument for `lowest_pass(passes)` and `highest_pass(passes)`, which are smaller single-purpose functions that return the relevant pass (or nil)

Would that work for this use case? Hopefully I'm not interpreting the intent incorrectly!

Reply:
Still trying to see if I'm with you yet or not.

When you called `lowest_pass(passes)`, would you be passing in the passes for all the legs? I think this would be a problem because you don't necessarily want the absolute lowest. In the case of the subway to CR trip, for all legs, you'd have 3 passes: the subway Monthly LinkPass, the CR CharlieTicket, and the CR mTicket. What we want to display in the accordion is the CR CharlieTicket and the CR mTicket. But the lowest of all these is the subway Monthly LinkPass. 

Or am I misunderstanding the idea? Would you just be passing in the passes for one leg or something different altogether?

Reply:
I definitely think the confusing names are still adding to the overall confusion in all of this!

Reply:
Ah great point about the lowest_pass, indeed part of my misunderstanding (which you did interpret correctly) was caused by the naming! That definitely makes my logic flawed.

As for naming, why are we annotating anything as "lowest" then? Definitely seems more confusing than useful here. I'd think we want mTicket in the display not because it's a "lowest" anything, but because it's another applicable pass for the trip (if there were ever a third CR pass, we'd want all three listed no?).